avr-hal-generic/src/usart_spi.rs

```diff
@@ -0,0 +1,91 @@
+//! MSPIM Implimentation
+use crate::spi;
+
+// This module just impliments a macro for SpiOps, since underlyingly, the Spi type can still be used since it just needs SpiOps
+
+pub type UsartSpi<H, USART, SCLKPIN, MOSIPIN, MISOPIN, CSPIN> =
+    spi::Spi<H, USART, SCLKPIN, MOSIPIN, MISOPIN, CSPIN>;
+
+// Impliment SpiOps trait for USART
+#[macro_export]
+macro_rules! impl_usart_spi {
+    (
+        hal: $HAL:ty,
+        peripheral: $USART_SPI:ty,
+        register_suffix: $n:expr,
+        sclk: $sclkpin:ty,
+        mosi: $mosipin:ty,
+        miso: $misopin:ty,
+        cs: $cspin:ty,
+    ) => {
+        $crate::paste::paste! {
+            impl $crate::spi::SpiOps<$HAL, $sclkpin, $mosipin, $misopin, $cspin> for $USART_SPI {
```

Comment:
Wow, didn't think you'd be able to get by with just implementing the existing `SpiOps`.  That's pretty neat :)

Reply:
I realised, underlyingly, `Spi` just utilises `SpiOps`, and could cheap out.

Either sometime in the next 3-4 hours, or sometimes in 14 hours, I will have the implementations for the rest of the boards.

Reply:
Haha, don't stress over it!

BTW, please check the failing CI jobs, seems there are still some kinks to iron out...